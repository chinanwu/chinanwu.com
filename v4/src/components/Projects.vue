<template>
  <section class="Projects" id="projects" aria-labelledby="projectsHeader">
    <header>
      <h2 id="projectsHeader" class="headerLabel">Projects</h2>
    </header>

    <ArrowNavTabs
      :items="items"
      :active="current"
      @selected="setCurrent"
      aria-labelledby="projectsHeader"
      aria-controls="project"
      :focusActive="focusActive"
    />
    <Project id="project" :name="current" aria-live="polite" />

    <div class="Projects__arrowSelectorContainer">
      <ArrowSelector
        :active="current"
        :items="items"
        @change="setCurrentFromArrow"
        aria-labelledby="projectsHeader"
        aria-controls="project"
      />
    </div>
  </section>
</template>

<script>
import ArrowNavTabs from "@/components/ArrowNavTabs";
import ArrowSelector from "@/components/ArrowSelector";
import Project from "@/components/Project";

export default {
  name: "Projects",
  components: { ArrowSelector, Project, ArrowNavTabs },
  data() {
    return {
      items: ["delta", "dnd", "pomelo-lib", "showcase"],
      current: "delta",

      // A bit of a hacky solution to the problem of arrow selector actions causing focus to jump to the nav menu
      focusActive: true
    };
  },
  methods: {
    setCurrent(next) {
      this.current = next;
      this.focusActive = true;
    },
    setCurrentFromArrow(nextNum) {
      this.current = this.items[nextNum - 1];
      this.focusActive = false;
    }
  }
};
</script>

<style lang="less" scoped>
.Projects__arrowSelectorContainer {
  display: flex;
  justify-content: center;
}
</style>
