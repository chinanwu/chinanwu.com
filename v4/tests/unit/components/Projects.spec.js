import { shallowMount } from "@vue/test-utils";
import Projects from "@/components/Projects.vue";

describe("Projects", () => {
  it("renders", () => {
    const wrapper = shallowMount(Projects);
    expect(wrapper.element).toMatchSnapshot();
  });
});
