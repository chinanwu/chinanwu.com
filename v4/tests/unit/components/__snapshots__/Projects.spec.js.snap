// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects renders 1`] = `
<section
  aria-labelledby="projectsHeader"
  class="Projects"
  id="projects"
>
  <header>
    <h2
      class="headerLabel"
      id="projectsHeader"
    >
      Projects
    </h2>
  </header>
  <arrow-nav-tabs-stub
    aria-controls="project"
    aria-labelledby="projectsHeader"
  />
  <project-stub
    aria-live="polite"
    id="project"
  />
  <div
    class="Projects__arrowSelectorContainer"
  >
    <arrow-selector-stub
      aria-controls="project"
      aria-labelledby="projectsHeader"
    />
  </div>
</section>
`;
